test/test-genesis.ts

```diff
@@ -1,28 +1,50 @@
-import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
-import { expect } from "chai";
-import { Contract, utils } from "ethers";
-import { ethers } from "hardhat";
-import { Deployment } from "hardhat-deploy/dist/types";
+/* eslint-disable no-unused-expressions */
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { expect } from 'chai';
+import { Contract, utils } from 'ethers';
+import { ethers } from 'hardhat';
+import { Deployment } from 'hardhat-deploy/dist/types';
 import {
   addLinkFundIfNeeded,
   addressZero,
   deployTestContract,
   mint,
-} from "./test-utils";
+} from './test-utils';
 
-describe("Genesis Contract", () => {
+describe('Genesis Contract', () => {
   let contract: Contract;
   let VRFCoordinatorMock: Deployment;
   let owner: SignerWithAddress;
-  let address1: SignerWithAddress;
   let oracle: SignerWithAddress;
+  let whitelisted: SignerWithAddress;
+  let notWhitelisted: SignerWithAddress;
   const randomNumber = 777;
+  const merkleTreeRoot =
+    '0xe32689cba5d4fa1698e45bcc090b47a13a1ea9c41671c5205bebeecd5b9ebb7f';
+  const nonce = 0;
+  const proof = [
+    '0x6979242712a952fdd0e6020e3b0c5ab00e441efc3a8e5f3550b44bdfef0bc8fe',
+    '0xc2fcc0de40e2633995715cba771542dd0e55202ea7203cba6597154fa5166b9a',
+    '0x7f5e7f7411eff0a9aa5209931feb717d400cdea255ab6b415205b35e0b4e5dd1',
+    '0xd0c7c1535456333e883f588cd7b1fd89c69d9cccb3e44384df48131d279c7242',
+    '0x917003bd175f847c8cace233737e6bea0ec6857553135fad5679bafa987377c6',
+    '0xf55e6ce46027df69d182f452137be4258598c22fd4c86ea842590ae9eb1aad7b',
+    '0x2ca4e96251485c7f5b3078c31b7befb7df302db3f70475be3850d200da04915f',
+    '0xf7a619404c2ecfbc206616a2be4a8c6d737a47e50c4f321c4bc4db2823d49a82',
+    '0x88dfe75f40c9b2f91343fa790ca8779dd027c5ed230ce71b83b56fbae4503c3b',
+    '0x4295d089ba50f6833e1eebf5dc495f3ab00c0ca99de72602439df04fd0fd00c4',
+  ];
+  const bogusNonce = 99999;
+  const bogusProof = [
+    '0x0ffde5c80d693e686066165e79e1aa33f44b9b3b61ab358e9cda2cfa5988c2af',
+  ];
 
-  beforeEach(async () => {
+  before(async () => {
```

Comment:
hmmm no, we want this done before each test no?

Reply:
no! exactly! we want to only deploy once before all the tests. Thats how I see it.

Reply:
Nope, why? Unless you want to have different test suites, but I think each test should be ran independently and should not rely on any other tests being ran before.

Use case: I am writing a new function and want to do it in TDD, I start writing my test, I have to go through all the test suite to mak sure it passes vs. just running my test. Also, that means I have to be careful of the order I do my tests. I don't think we want that at all.